libs/agno/agno/agent/agent.py

```diff
@@ -1577,9 +1578,23 @@ def run(
         )
 
         # Read existing session from database
-        agent_session = self._read_or_create_session(session_id=session_id, user_id=user_id)
+        agent_session = None
+        if session is not None:
+            agent_session = session
+        else:
+            agent_session = self._read_or_create_session(session_id=session_id, user_id=user_id)
+
+        if agent_session is None:
```

Comment:
We already have this check in read_or_create_session I believe, because it would create one